src/Doctrine/DoctrineHelper.php

```diff
@@ -271,6 +272,19 @@ public static function getPropertyTypeForColumn(string $columnType): ?string
             'ulid' => '\\'.Ulid::class,
             default => null,
         };
+
+        if (null === $propertyType && Type::getTypeRegistry()->has($columnType)) {
+            $reflection = new \ReflectionClass(\get_class(Type::getTypeRegistry()->get($columnType)));
+            if ($reflection->hasMethod('convertToPHPValue') && $returnType = $reflection->getMethod('convertToPHPValue')->getReturnType()) {
```

Comment:
We should be able to drop this conditional all together. `getType()` always returns a `Type` and abstract `Type` has the convert method.

https://github.com/doctrine/dbal/blob/f3defbb4f88a7472314746f122f6d4d84615d424/src/Types/TypeRegistry.php#L41